pkg/apis/cr/v1alpha1/types.go

```diff
@@ -248,13 +248,15 @@ type Location struct {
 type CredentialType string
 
 const (
-	CredentialTypeKeyPair CredentialType = "keyPair"
+	CredentialTypeKeyPair     CredentialType = "keyPair"
+	CredentialTypeTypedSecret CredentialType = "typedSecret"
```

Comment:
Will it be possible to use the `secret` field inside `KeyPair`?

Reply:
Don't we want a good separation between these two?
Do you think add a new union member struct is bad for us?

Reply:
Just thinking of ways to make it generic. Will the secret contain just the role or even the AWS key and ID?

Reply:
Got it. It will have only secret field. The reason I am trying to make it separate is that the consumer code should understand which type credentials they are by looking at the type. Then it should decode the data in the secret reference with a know logic. For example in this case, if it is typed secret, then fetch the secret with its reference. Then check the type of the secret. If it is a AWS typed secret, then fetch the well-known keys from its data section.

Reply:
Okay, thanks for the clarification.